services/translateService.js

```diff
@@ -42,40 +29,29 @@ const setTranslateWithFallbackEngines = ({
   });
   const enginesFailed = [];
 
-  /**
-   * Translates the given text from one language to another using the specified translation engines in order of preference.
-   * @param {string} text - The text to be translated.
-   * @param {string} from - The language code of the text to be translated.
-   * @param {string} to - The language code to translate the text to.
-   * @param {string[]} engines - An array of translation engines to use, in order of preference.
-   * @returns {Promise<{text: string}>} - A Promise that resolves to an object containing the translated text.
-   * @throws {Error} - If there is not translation result.
-   */
   const translateWithFallbackEngines = async (text, from, to) => {
     let result;
 
-    // Avoid trying to use engines that have failed in the past to save time and network requests
     const enginesFiltered = engines.filter(
-      (engine) => !enginesFailed.includes(engine),
+      (engine) => !enginesFailed.includes(engine.name),
     );
 
     for await (const engine of enginesFiltered) {
       try {
-        // Validate that the language is supported by the engine to avoid unnecessary network requests
```

Comment:
I added this comment because it could be confusing since we are inside a loop and it could be complex to read for the first time, maybe we should preserve it